### ISSUE TITLE
Enable text editing of my answers

### ISSUE THREAD

**Issue by tedbarnett** [tedbarnett/Boswell#79]
Not critical, but will be nice to have:

OpenAI (or maybe it's Siri) often makes mistakes when interpreting my speech. Before sending off the chat conversation to my friends (and before we make video-shares possible), the user might want to correct these typos!

If user taps-and-holds on any of his own text (ie stuff I said, but not what the AI said), then an Edit/Cancel option pops up. If I click "edit", then the keyboard appears and I can edit the text of that response. Cancel/Save buttons appear at the bottom of the screen.

**Comment by bhautikpatoliya004** [comment]
@tedbarnett  Yes, editing his own text is a good idea. but the problem is if the user updates his own text then we can't update his audio speech because as you know we can't record updated text audio without user interaction.

if we want only update text then I will do it. please let me know your thoughts on it

**Comment by tedbarnett** [comment]
Yes: just the text itself.  No need to touch the audio from the user.  I realize that the user might edit the text so that it looks nothing like the audio, but that's okay.

If the speech-to-text was more accurate, we wouldn't need this, but I'm not sure we can ever solve that perfectly.  People use unusual nouns (e.g. names, places, etc.) that no speech-to-text system will ever get right.

What is interesting and weird is that sometimes the text looks wrong (e.g. thinks I worked at "McKenzie and Company"), but then ChatGPT will correctly refer to it as "McKinsey & Company".  AI is amazing.  I guess it's doing its own spell and grammer-fixing even if we feed it errors!
